js/items/building.js

```diff
@@ -0,0 +1,76 @@
+class Building {
+  constructor(poly, heightCoef = 0.4) {
+    this.base = poly;
+    this.heightCoef = heightCoef;
+  }
+  draw(ctx, viewPoint) {
+    //calculate the top part of the building using view point
+    const topPoints = this.base.points.map((p) =>
+      add(p, scale(subtract(p, viewPoint), this.heightCoef * 0.6))
```

Comment:
by multiplying the heightCoef by 0.6, making the topPoints(later will be ceiling in line #49) lower than the topMidPoint in line #45)